app/services/comment_viewers_service.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+class CommentViewersService
+  attr_reader :comment, :owner, :current_user, :commentable_viewers
+
+  def self.viewers(comment, current_user)
+    new(comment, current_user).viewers
+  end
+
+  def self.viewable(comment, current_user)
+    new(comment, current_user).viewable
+  end
+
+  def self.deletable(comment, current_user)
+    new(comment, current_user).deletable
+  end
+
+  def initialize(comment, current_user)
+    commentable = get_commentable(comment)
+    @comment = comment
+    @owner = commentable[:user_id] && User.find(commentable[:user_id])
+    @commentable_viewers = commentable[:viewers] ||
```

Comment:
Using `if !commentable.nil?` instead, since using exists? caused a problem i.e. `undefined method `exists?' for #<Moment:0x007fa06e7cc458>`. Great catch!